<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find exploration)/cfg/custom_rosconsole.config"/>

    <arg name="robot_name" default="robot1"/>

    <arg name="map" default="/$(arg robot_name)/map"/>
    <!-- <arg name="map" default="/server/grid_map_merge/merge_map"/> -->
    <arg name="pose" default="/$(arg robot_name)/robot_pose"/>

    <arg name="velocity" default="/$(arg robot_name)/mobile_base/commands/velocity"/>
    <arg name="scan" default="/$(arg robot_name)/scan"/>
    <arg name="bumper_raw" default="/$(arg robot_name)/mobile_base/events/bumper"/>
    <arg name="map_frame_id" default="$(arg robot_name)/map"/>

    <arg name="rosbag" default="true"/>
    <arg name="debug_mode" default="false">


    <arg name="ns" default="exploration"/>

    <group ns="$(arg robot_name)/$(arg ns)">
        <node pkg="exploration" type="frontier_based_exploration" name="frontier_based_exploration" output="screen">
            <remap from="map" to="$(arg map)"/>
            <remap from="pose" to="$(arg pose)"/>
            <param name="map_frame_id" value="$(arg map_frame_id)"/>
            <param name="movebase_name" value="/$(arg robot_name)/move_base"/>
            <param name="debug_mode" value="$(arg debug_mode)"/>
            <!-- <param name="merge_map_frame_id" value="/server/merge_map"/> -->
            <!-- <param name="use_merge_map" value="true"/> -->
        </node>

        <node pkg="exploration" type="visualization" name="visualization">
            <param name="map_frame_id" value="$(arg robot_name)/map"/>
        </node>

        <node pkg="exploration" type="log" name="log">
            <remap from="pose" to="$(arg pose)"/>
        </node>

	<include if="$(arg rosbag)" file="$(find exploration)/launch/includes/rosbag_record.launch.xml">
            <!-- <arg name="dir" value="/home/ros1/rosbag_debug"/> -->
            <arg name="topics" value="  $(arg pose) 
                                        pose_log 
                                        $(arg bumper_raw) 
                                        $(arg scan) 
                                        $(arg velocity) 
                                        visualization 
                                        /tf 
                                        /tf_static 
                                        /$(arg robot_name)/map 
                                        /$(arg robot_name)/move_base/global_costmap/costmap 
                                        /$(arg robot_name)/NavfnROS/plan"/>
        </include>

    </group>
</launch>
